json/signin.js

```diff
@@ -1,70 +1,51 @@
+import {testUser, emailValidCheck, addError, removeError, passwordToggle} from "./global.js";
 const emailInput = document.querySelector("#email");
+const emailError = document.querySelector(".email-error");
 const passwordInput = document.querySelector("#password");
+const passwordError = document.querySelector(".password-error");
 const confirmBtn = document.querySelector('#confirm-button');
-const email_regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
+const eyeBtn = document.querySelector('.eye-btn');
 
-/*email 형식인지 체크해주는 함수*/
-function emailCheck(email_address){     
-	if (!email_regex.test(email_address)) { 
-		return false; 
-	} else {
-		return true;
-	}
-}
 
 /*입력받은 email 값을 판별하고 조건에 맞는 에러메시지를 출력하는 함수*/
-function checkInputEmail(event) {
-  const errorMessage = document.createElement('div');
-  errorMessage.classList.add('error-message');
-  emailInput.classList.add('error-border');
-  if (event.target.value === '') {
-    errorMessage.innerHTML = '<p>이메일을 입력해 주세요.</p>';
-    emailInput.after(errorMessage);
-  } else if (!emailCheck(event.target.value)) {
-    errorMessage.remove();
-    errorMessage.innerHTML = '<p>올바른 이메일 주소가 아닙니다.</p>';
-    emailInput.after(errorMessage);
-  } else {
-    errorMessage.style.display = 'none';
-    emailInput.classList.remove('error-border');
+function validEmailInput(e) {
+  if (e.target.value === '') {
+    addError(emailInput, emailError, '이메일을 입력해 주세요.');
+    return;
+  } 
+  if (!emailValidCheck(e.target.value)) {
+    addError(emailInput, emailError, '올바른 이메일 주소가 아닙니다.');
+    return;
```

Comment:
input의 값에 공백만 있을 경우를 대비하면 좋겠네요.

```suggestion
function validEmailInput(e) {
  const emailValue = e.target.value.trim();
  
  if (emailValue === '') {
    addError(emailInput, emailError, '이메일을 입력해 주세요.');
    return;
  } 
  if (!emailValidCheck(emailValue)) {
    addError(emailInput, emailError, '올바른 이메일 주소가 아닙니다.');
    return;
```